src/middlewared/middlewared/plugins/docker_linux/client.py

```diff
@@ -58,14 +58,17 @@ async def _get_manifest_response(self, registry, image, tag, headers, mode, rais
             f'https://{registry}/v2/{image}/manifests/{tag}', headers=headers, mode=mode
         )
         if raise_error and response['error']:
-            raise CallError(f'Unable to retrieve latest image digest for {f"{image}:{tag}"!r}: {response["error"]}')
+            raise CallError(f"Unable to retrieve latest image digest for registry={registry} "
+                            f"image={image} tag={tag}: {response['error']}")
 
         return response
 
     @private
     async def get_manifest_call_headers(self, registry, image, headers):
         if registry == DEFAULT_DOCKER_REGISTRY:
             headers['Authorization'] = f'Bearer {await self._get_token(image)}'
+        else:
+            headers['Authorization'] = f'Bearer {await self._get_token(f"{registry}/{image}")}'
```

Comment:
Would this be required for custom registries ? `_get_token` gets token from `https://auth.docker.io`

Reply:
I'd say nice catch there!

Reply:
Scope has further increased as I followed through docker auth specs, we need to be able to get auth endpoint and scopes from response headers when we get 401 while fetching manifests: https://docs.docker.com/registry/spec/auth/token/

+ I need to add support for 2 schema versions that we specify in our headers while we request: https://docs.docker.com/registry/spec/manifest-v2-2/